bdsf/plotresults.py

```diff
@@ -247,7 +247,12 @@ def plotresults(img, ch0_image=True, rms_image=True, mean_image=True,
         numx = 1
     numy = int(N.ceil(float(len(images))/float(numx)))
     fig = pl.figure(figsize=(max(15, 10.0*float(numy)/float(numx)), 10.0))
-    fig.canvas.set_window_title('PyBDSM Fit Results for '+ img.filename)
+    try:
+        # This way has been deprecated with Matplotlib 3.4.
+        fig.canvas.set_window_title('PyBDSM Fit Results for '+ img.filename)
+    except AttributeError:
+        # We are on a newer Matplotlib where the above is deprecated.
+        fig.canvas.manager.set_window_title('PyBDSM Fit Results for '+ img.filename)
```

Comment:
I would prefer the logic to be reversed. In the sense that the happy path is with latest Matplotlib, and the unhappy path with old versions. Or..., assuming we can easily check the version of Matplotlib, use an `if/else` construct.

Reply:
The new behavior already works in matplotlib 1.2.0, released 2021, and the default in centos 7. I suggest to just use the new way and get rid of the try/except.

Reply:
Right, if it has been in there that long already I would assume most people are somewhat up to date and that the new way is probably safe. I've changed it to the new way in that case.

Reply:
Fine with me. Please add a minimal version requirement for `matplotlib` in `setup.py`.
Oh, I see that `matplotlib` doesn't even appear at all in the `install_requires`. Not sure why. Probably because it's only needed when using the extra `[ishell]`. I always find `pip install` and `setup.py` a bit awkward when it comes to extras. So, maybe just leave `setup.py` as is.